### PR TITLE
Adding import_options module from FMI2 to FMI3

### PR DESCRIPTION
**Pull request by PeterMeisrimelModelon**

### CODE REVIEW

File: Test/FMI3/fmi3_import_options_test.cpp [hunk 1]
```diff
@@ -141,31 +122,21 @@ int main(int argc, char *argv[])
...
-
-    fmu = fmi3_import_parse_xml(context, tmpPath, NULL);
+    REQUIRE(version == fmi_version_3_0_enu);
 
-    if (!fmu) {
-        printf("Error parsing XML. Exiting.\n");
-        do_exit(CTEST_RETURN_FAIL);
-    }    
+    fmu = fmi3_import_parse_xml(context, tmpPath, nullptr);
+    REQUIRE(fmu != nullptr);
```
Comment (MikaelJohnssonModelon) [inline Test/FMI3/fmi3_import_options_test.cpp:139]: Great, looks like this simplify the tests a bit.  How do the error messages looks from this? Do REQUIRE generate good assert failed messages?

File: src/Import/src/FMI3/fmi3_import_capi.c [hunk 2]
```diff
@@ -29,6 +29,18 @@ extern "C" {
 
 static const char * module = "FMILIB";
 
+
+static void fmi3_import_capi_destroy_dllfmu_and_restore_options(fmi3_import_t* fmu) {
+	if (fmu->capi->options) {
+		/* Take back ownership of Options */
```
Comment (modelonrobinandersson) [inline src/Import/src/FMI3/fmi3_import_capi.c:35]: I think the indentation in this function needs some adjustment.

File: src/Import/src/FMI3/fmi3_import_capi.c [hunk 3]
```diff
@@ -104,14 +116,19 @@ jm_status_enu_t fmi3_import_create_dllfmu(fmi3_import_t* fmu, fmi3_fmu_kind_enu_
                 logMessageFinal, fmuKind);
     }
 
+	if (fmu->capi) {
+		/* Replace the CAPI options with the import ones */
```
Comment (modelonrobinandersson) [inline src/Import/src/FMI3/fmi3_import_capi.c:120]: I think the indentation needs some adjustment.

File: src/Import/src/FMI3/fmi3_import_capi.c [hunk 4]
```diff
@@ -104,14 +116,19 @@ jm_status_enu_t fmi3_import_create_dllfmu(fmi3_import_t* fmu, fmi3_fmu_kind_enu_
...
+        fmi_util_free_options(fmu->callbacks, fmu->capi->options);
+		fmu->capi->options = fmu->options;
```
Comment (modelonrobinandersson) [inline src/Import/src/FMI3/fmi3_import_capi.c:122]: Here as well

**Review by modelonrobinandersson — APPROVED** [review]
Minor nitpick

File: src/Import/src/FMI3/fmi3_import.c [hunk 5]
```diff
@@ -55,6 +55,13 @@ fmi3_import_t* fmi3_import_allocate(jm_callbacks* cb) {
         return NULL;
     }
 
+	fmu->options = fmi_util_allocate_options(cb);
+	if (!fmu->options) {
+        fmi3_xml_free_model_description(fmu->md);
+        cb->free(fmu);
+        return NULL;
```
[the remark this answers is not in the document]
Comment (PeterMeisrimelModelon) [reply src/Import/src/FMI3/fmi3_import.c:62]: Good catch, these were some missing tabs to spaces conversions

File: src/Import/src/FMI3/fmi3_import_capi.c (hunk 2 above)
Comment (PeterMeisrimelModelon) [reply src/Import/src/FMI3/fmi3_import_capi.c:35]: Fixed

File: src/Import/src/FMI3/fmi3_import_capi.c (hunk 3 above)
Comment (PeterMeisrimelModelon) [reply src/Import/src/FMI3/fmi3_import_capi.c:120]: Fixed.

File: src/Import/src/FMI3/fmi3_import_capi.c (hunk 4 above)
Comment (PeterMeisrimelModelon) [reply src/Import/src/FMI3/fmi3_import_capi.c:122]: Fixed.

File: Test/FMI3/fmi3_import_options_test.c [hunk 6]
```diff
@@ -69,7 +68,7 @@ void test_option_memory_management(fmi3_import_t* fmu, fmi3_logger_context_t* cb
     fmi_import_set_option_loadlibrary_flag(opts, flag);
 
     /* Test after creating dllfmu */
-    status = fmi3_import_create_dllfmu(fmu, fmi3_fmu_kind_me, cbf->instanceEnvironment, cbf->logMessage);
```
[the remark this answers is not in the document]
Comment (MikaelJohnssonModelon) [reply Test/FMI3/fmi3_import_options_test.c]: Ok, Just looked wired

File: Test/FMI3/fmi3_import_options_test.cpp (hunk 1 above)
Comment (MikaelJohnssonModelon) [reply Test/FMI3/fmi3_import_options_test.cpp:139]: okey, good then.

**Pull request merged**